generators/gradle/index.cjs

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2013-2021 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable consistent-return */
+const chalk = require('chalk');
+const { mixBlueprintGenerator } = require('generator-jhipster/support');
+
+const { GENERATOR_PROJECT_NAME, GENERATOR_JAVA, GENERATOR_GRADLE } = require('../generator-list');
+const { files } = require('./files.cjs');
+const { GRADLE, GRADLE_VERSION } = require('./constants.cjs');
+const { BUILD_TOOL } = require('../java/constants.cjs');
+
+const MixedGenerator = mixBlueprintGenerator(GENERATOR_PROJECT_NAME, GENERATOR_JAVA);
+
+module.exports = class extends MixedGenerator {
+  constructor(args, opts, features) {
+    super(args, opts, { jhipsterModular: true, unique: 'namespace', ...features });
+
+    // Register options available to cli.
+    if (!this.fromBlueprint) {
+      this.registerCommonOptions();
+      this.registerProjectNameOptions();
+    }
+
+    if (this.options.help) return;
+
+    this.config.defaults({
```

Comment:
Couldn't it be a step in the yo queue?

Reply:
It's better at the constructor.
Defaults are better to be set before starting the queue.
Any reason to move to a queue task?

Reply:
1. modifying the defaults depending on the generator call chain
e.g. if I choose init->java->spring->gateway, I want to have the `reactive` default value of the spring generator set to `true` because I selected the gateway generator
2. to be able to easily configure a company blueprint, that will override some of the jhipster defaults to match my company stack and standards

Reply:
I will try that.

Reply:
It won't work.
Tasks can run with the order inverted, so generator-java will have it's defaults before generator-spring-boot.
So generator-spring-boot defaults won't take priority over generator-java defaults.